monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -102,39 +113,56 @@ def get_tunnels():
+        machines = ReportService.get_all_machines()
 
-        for node in nodes:
+        for machine in machines:
             # This information should be evident from the map, not sure a table/list is a good way
             # to display it anyways
-            nodes_that_can_access_current_node = node["accessible_from_nodes_hostnames"]
+            addresses = [str(iface.ip) for iface in machine.network_interfaces]
+            accessible_machines = [
+                m.hostname for m in ReportService.get_accessible_machines(machine)
+            ]
             formatted_nodes.append(
                 {
-                    "label": node["label"],
-                    "ip_addresses": node["ip_addresses"],
-                    "accessible_from_nodes": nodes_that_can_access_current_node,
-                    "services": node["services"],
-                    "domain_name": node["domain_name"],
-                    "pba_results": node["pba_results"] if "pba_results" in node else "None",
+                    "label": machine.hostname,
+                    "ip_addresses": addresses,
+                    "accessible_from_nodes": accessible_machines,
+                    "services": [],
+                    "domain_name": "",
+                    "pba_results": "None",
                 }
             )
 
         logger.info("Scanned nodes generated for reporting")
 
         return formatted_nodes
 
-    @staticmethod
-    def get_all_displayed_nodes():
-        nodes_without_monkeys = [
-            NodeService.get_displayed_node_by_id(node["_id"], True)
-            for node in mongo.db.node.find({}, {"_id": 1})
-        ]
-        nodes_with_monkeys = [
-            NodeService.get_displayed_node_by_id(monkey["_id"], True)
-            for monkey in mongo.db.monkey.find({}, {"_id": 1})
-        ]
-        nodes = nodes_without_monkeys + nodes_with_monkeys
-        return nodes
+    @classmethod
+    def get_accessible_machines(cls, machine: Machine):
+        if cls._node_repository is None or cls._machine_repository is None:
+            return []
+
+        nodes = cls._node_repository.get_nodes()
+        machine_iter = (node for node in nodes if node.machine_id == machine.id)
+        accessible_machines = set()
+        for source in machine_iter:
+            for dest, conn in source.connections.items():
+                if CommunicationType.SCANNED in conn:
+                    accessible_machines.add(dest)
+
+        return [cls._machine_repository.get_machine_by_id(id) for id in accessible_machines]
+
+    @classmethod
+    def get_all_machines(cls) -> Iterable[Machine]:
+        if cls._machine_repository is None:
+            return iter(())
+        machines = cls._machine_repository.get_machines()
+        t1, t2 = tee(machines)
+
+        def is_island(machine: Machine):
+            return machine.island
+
+        return chain(filter(is_island, t1), filterfalse(is_island, t2))
```

Comment:
So we duplicate the list of machine, then on one list we only leave the islands, on another one we only leave the non-islands. Then we concatenate these lists. The end result is the list we had in the first place?

Reply:
By "duplicate" I meant deep copy, not code duplication. I've tested this method and I didn't see the point. 1. There's no indication that the intent of this method is to sort machines by `is_island`. 2. If it's a sort method, then why not use sort function? 3. Sorting data is responsibility of the display (UI).